### PR TITLE
Feat/en 3302 wallet integration test refactor

### PR DESCRIPTION
**Pull request by bogdan-rosianu**
-removed testInitializer.go from wallet package
-used testProcessorNode.go instead

### CODE REVIEW

File: integrationTests/frontend/wallet/txInterception_test.go [hunk 1]
```diff
@@ -83,8 +81,6 @@ func TestInterceptedTxFromFrontendGeneratedParamsGasPriceGasLimitNoData(t *testi
 	)
 }
 
-// testInterceptedTxFromFrontendGeneratedParams tests that a frontend generated tx will pass through an interceptor
-// and ends up in the datapool, concluding the tx is correctly signed and follows our protocol
 func testInterceptedTxFromFrontendGeneratedParams(
```
Comment (BeniaminDrasovean) [inline integrationTests/frontend/wallet/txInterception_test.go]: Why did you remove the comment? Please put it back.

File: integrationTests/frontend/wallet/txInterception_test.go [hunk 2]
```diff
@@ -100,32 +96,16 @@ func testInterceptedTxFromFrontendGeneratedParams(
...
-	nodePubKeyBytes, _ := sk.GeneratePublic().ToByteArray()
-	nodeAddress, _ := addrConverter.CreateAddressFromPublicKeyBytes(nodePubKeyBytes)
-	nodeAccount, _ := accntAdapter.GetAccountWithJournal(nodeAddress)
-	_ = nodeAccount.(*state.Account).SetNonceWithJournal(startingNonce)
-	_, _ = accntAdapter.Commit()
-
 	chDone := make(chan struct{})
 
-	var err error
+	node := integrationTests.NewTestProcessorNode(1, 0, uint32(0), "nodeAddr")
```
Comment (BeniaminDrasovean) [inline integrationTests/frontend/wallet/txInterception_test.go]: You can remove uint32().

File: integrationTests/frontend/wallet/txInterception_test.go (hunk 1 above)
Comment (bogdan-rosianu) [reply integrationTests/frontend/wallet/txInterception_test.go]: yes, done

File: integrationTests/frontend/wallet/txInterception_test.go (hunk 2 above)
Comment (bogdan-rosianu) [reply integrationTests/frontend/wallet/txInterception_test.go]: done

File: integrationTests/frontend/wallet/txInterception_test.go [hunk 3]
```diff
@@ -100,32 +98,16 @@ func testInterceptedTxFromFrontendGeneratedParams(
...
-
-	addrConverter, _ := addressConverters.NewPlainAddressConverter(32, "0x")
-	accntAdapter := createAccountsDB()
-
-	shardCoordinator := &sharding.OneShardCoordinator{}
-
-	n, _, sk, _ := createNetNode(dPool, accntAdapter, shardCoordinator)
-
-	//set the account's nonce to startingNonce
-	nodePubKeyBytes, _ := sk.GeneratePublic().ToByteArray()
```
Comment (BeniaminDrasovean) [inline integrationTests/frontend/wallet/txInterception_test.go]: You could add 
```go
// SetAccountNonce sets the account nonce with journal
func (tpn *TestProcessorNode) SetAccountNonce(nonce uint64) {
	nodePubKeyBytes, _ := tpn.SkTxSign.GeneratePublic().ToByteArray()
	nodeAddress := CreateAddresFromAddrBytes(nodePubKeyBytes)
	nodeAccount, _ := tpn.AccntState.GetAccountWithJournal(nodeAddress)
	_ = nodeAccount.(*state.Account).SetNonceWithJournal(nonce)
	_, _ = tpn.AccntState.Commit()
}
```
 to the TestProcessorNode, and use the method here

Comment (bogdan-rosianu) [reply integrationTests/frontend/wallet/txInterception_test.go]: done. also treated errors

File: integrationTests/frontend/wallet/txInterception_test.go [hunk 4]
```diff
@@ -100,32 +98,19 @@ func testInterceptedTxFromFrontendGeneratedParams(
...
-	nodeAddress, _ := addrConverter.CreateAddressFromPublicKeyBytes(nodePubKeyBytes)
-	nodeAccount, _ := accntAdapter.GetAccountWithJournal(nodeAddress)
-	_ = nodeAccount.(*state.Account).SetNonceWithJournal(startingNonce)
-	_, _ = accntAdapter.Commit()
-
 	chDone := make(chan struct{})
 
-	var err error
+	node := integrationTests.NewTestProcessorNode(1, 0, 0, "nodeAddr")
```
Comment (sasurobert) [inline integrationTests/frontend/wallet/txInterception_test.go]: extrat 1, 0, 0, "nodeAddr" into variables.

Comment (bogdan-rosianu) [reply integrationTests/frontend/wallet/txInterception_test.go]: done
